Automattic/jetpack
# Move wpcom/v2/publicize/connections endpoint to publicize package

## Proposed changes:

* Move `wpcom/v2/publicize/connections` endpoint to publicize package that can be used in every scenario the publicize module is used
    * WPCOM sites
    * Jetpack sites - both Jetpack and Social plugins

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Goto Jetpack admin or social admin page to verify connections management loads the connections fine.
* Apply this PR to your sandbox and point public API to the sandbox
* Goto [Developer console](https://developer.wordpress.com/docs/api/console/)
* Hit WP REST API - `wpcom/v2/sites/<site_id>/publicize/connections`
* Confirm that the site connections are displayed correctly as per the schema in this PR
* Break a connection by removing Jetpack from the service, e.g. [here for Facebook](https://www.facebook.com/settings/?tab=business_tools).
* Add `?test_connections=1` to the endpoint above
* Verify that the tests are run for connections
* Verify that `status` is set to `'broken'` for the broken connection
* Mark a connection as global in Calypso connections screen
* Verify that global is set to true for backwards compatibility
* Point your Jetpack site to the sandbox
* Goto any page which uses `wp.apiFetch` e.g. Block Editor
* In console, run this
```js
await wp.apiFetch({
    path: '/wpcom/v2/publicize/connections'
});
```
* Confirm that the site connections are displayed correctly as per the schema in this PR

<img width="1022" alt="Screenshot 2024-12-13 at 10 27 05 AM" src="https://github.com/user-attachments/assets/aa55930f-177e-4289-b1d1-026933bd2b70" />
